pkgs/development/python-modules/outlines-core/default.nix

```diff
@@ -0,0 +1,78 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  cargo,
+  rustPlatform,
+  rustc,
+  setuptools-rust,
+  setuptools-scm,
+  interegular,
+  jsonschema,
+  datasets,
+  numpy,
+  pytestCheckHook,
+  pydantic,
+  scipy,
+  torch,
+  transformers,
+}:
+
+buildPythonPackage rec {
+  pname = "outlines-core";
+  version = "0.1.17";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "outlines_core";
+    hash = "sha256-CSWhKrHogYxvGJb+79tNTYySLmRhUVBi0b0+YEWbAoE=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  postPatch = ''
+    ls -l
+    cat ${./Cargo.lock} > Cargo.lock
+    ls -l
+  '';
+
+  pyproject = true;
+
+  build-system = [
+    cargo
+    rustPlatform.cargoSetupHook
+    rustc
+    setuptools-rust
+    setuptools-scm
+  ];
+
+  dependencies = [
+    interegular
+    jsonschema
+  ];
+
```

Comment:
```suggestion

  optional-dependencies = {
    test = [
      pre-commit
      pydantic
      pytest
      pytest-benchmark
      pytest-cov-stub
      pytest-mock
      coverage
      diff-cover
      accelerate
      beartype
      huggingface-hub
      torch
      numpy
      scipy
      transformers
      datasets
      pillow
      asv
      setuptools-rust
    ];
  };

```

Reply:
The set of dependencies was hand-checked. Upstream has a bunch of dependencies in there that are not even used anywhere in the project. Also, some of the dependencies are only used in benchmarks and not the test suite.

Do we still want to add test dependencies that are not used?

Done: put the Python test dependencies in `optionalDependencies.test`.

Reply:
It's *probably* be fine, but to be on the safe side (and avoid needing to manually pick through what's being used again in the future) I would continue to use what upstream declares